screenpy_examples/screenpy/quietly_logging/actions/do_thing.py

```diff
@@ -0,0 +1,100 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from screenpy import IsEqual, Performable, beat
+
+from screenpy_examples.screenpy.quietly_logging.actions.see import See
+from screenpy_examples.screenpy.quietly_logging.questions import SimpleQuestion
+
+if TYPE_CHECKING:
+    from screenpy import Actor
+
+
+class DoPassAfterAWhile(Performable):
+    LIMIT = 5
+    COUNTER = 0
+
+    @beat("{} tries to DoPassAfterAWhile")
+    def perform_as(self, actor: Actor):
+        if DoPassAfterAWhile.COUNTER > DoPassAfterAWhile.LIMIT:
+            DoPassAfterAWhile.COUNTER = 0
+            return
+
+        DoPassAfterAWhile.COUNTER += 1
+        raise AssertionError(f"This is going to Fail {DoPassAfterAWhile.COUNTER}")
+
+
+class DoFailCounter(Performable):
+    COUNTER = 0
+
+    @beat("{} tries to DoFailCounter")
+    def perform_as(self, actor: Actor):
+        DoFailCounter.COUNTER += 1
+        raise AssertionError(f"This is going to Fail {DoFailCounter.COUNTER}")
```

Comment:
Maybe:
```python
plural = "s" if self.COUNTER == 1 else ""
msg = f"This has failed {self.COUNTER} time{plural}."
raise AssertionError(msg)
```

Reply:
I added the counter to show the different exceptions in the eventually stack. The point wasn't to show how many times it failed but rather the fact that each failure appears in the stack.

```
screenpy.exceptions.DeliveryError: Marcel tried to Eventually perform fail counter 10 times over 1 seconds, but got:
    AssertionError: This is going to Fail 1
    AssertionError: This is going to Fail 2
    AssertionError: This is going to Fail 3
    AssertionError: This is going to Fail 4
    AssertionError: This is going to Fail 5
    AssertionError: This is going to Fail 6
    AssertionError: This is going to Fail 7
    AssertionError: This is going to Fail 8
    AssertionError: This is going to Fail 9
    AssertionError: This is going to Fail 10

```